### PR TITLE
Dfe 463 - Methodology changes

### PR DESCRIPTION
**Pull request by negatron99**
Includes merge from master for common framework.

### CODE REVIEW

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import debounce from 'lodash/debounce';
-import React, { Component, KeyboardEvent } from 'react';
+import React, {ChangeEvent, Component, KeyboardEvent} from 'react';
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: Run Prettier before commits (should be running on precommit Git hook). Formatting problems gets picked up by the build.

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ interface State {
 }
 
 class PrototypeSearchForm extends Component<{}, State> {
-  public state = {
+  public state: State = {
     currentlyHighlighted: undefined,
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: Shouldn't have to define state as undefined as we can just avoid including it in the state object.

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx [hunk 3]
```diff
@@ -151,6 +191,16 @@ class PrototypeSearchForm extends Component<{}, State> {
 
       this.setState({ currentlyHighlighted });
     }
+
+    if (e.keyCode === 13) {
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: Use `e.key` as this is more ergonomic/readable than `e.keyCode` numbers

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx [hunk 4]
```diff
@@ -125,9 +114,60 @@ class PrototypeSearchForm extends Component<{}, State> {
...
+      searchValue: '',
+    });
+  }
+
+  private static calculateLocationOfElement(
+    element: HTMLElement,
+    insideAccordion: boolean,
+    potentialAccordion: HTMLElement,
+  ) {
+    const location = [];
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: Probably should have `string[]` type

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx [hunk 5]
```diff
@@ -125,9 +114,60 @@ class PrototypeSearchForm extends Component<{}, State> {
...
+          PrototypeSearchForm.substring(accordionHeader.textContent || '', 20),
+        );
+      }
+    }
+
+    return location;
+  }
+
+  private handleTextChange() {
+    debounce(() => {
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: Does this actually debounce properly? I would have thought you'd have to assign the debounced function into a class property

Comment (negatron99) [reply src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: I merged the debounce changes in from the master branch. I didn't have the debounce function in there are first.

File: src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx (hunk 3 above)
Comment (negatron99) [reply src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx]: String comparisons though?

**Comment by negatron99** [comment]
I've got to merge this to get the changes into master

**Pull request merged**